### PR TITLE
Update android.md to use NDK r26c

### PR DESCRIPTION
**Pull request by janisozaur**
Latest NDK is r26b, released 2023-10-11.

https://github.com/android/ndk/releases/tag/r26b

I have done a basic test by performing jsoncpp compilation with updated image.

### CODE REVIEW

**Comment by dg0yt** [comment]
Note that all autotools-based ports are currently broken with NDK r26, the proposed fix is rejected, and the desired more fundamental transformation hasn't arrived yet. Cf. https://github.com/microsoft/vcpkg/pull/31228

So I would recommend to not promote the use of vcpkg with NDK r26, or there will be a flood of more issues.
(Or maybe it needs these issue to force a solution from the vcpkg maintainers.)

**Comment by janisozaur** [comment]
I see. Does this affect r26**b** as well? I have just compiled libpng and it didn't report any errors with r26b, is that expected (should it fail at compile time or will it only fail later when trying to link/use at runtime)?

The PR you posted has commits that target vcpkg itself, is the fix expected on your side or is it something NDK needs to fix? In other words: Should this PR be closed until r27 is available or just remain blocked on 31228?

**Comment by dg0yt** [comment]
Test a port which uses `vcpkg_configure_make`. e.g. libb2, https://github.com/microsoft/vcpkg/issues/31332.

**Comment by janisozaur** [comment]
It failed - I see what should be expected for autoconf. In that case I understand the fix is to be implemented on vcpkg side.

I will leave this PR be until 31228 gets resolved. That said, thankfully none of packages I'm interested in uses autotools, I'm going to use NDK r26b.

For documentation's sake, here are the logs:
```
Building libb2:x64-android...
-- Installing port from location: /vcpkg/buildtrees/versioning_/versions/libb2/7c113e12089453e4e2cf2bbf67ad1f0b80a133a8
-- Using cached BLAKE2-libb2-2c5142f12a2cd52f3ee0a43e50a3a76f75badf85.tar.gz.
-- Cleaning sources at /vcpkg/buildtrees/libb2/src/6f75badf85-b3a5d32c8d.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /vcpkg/downloads/BLAKE2-libb2-2c5142f12a2cd52f3ee0a43e50a3a76f75badf85.tar.gz
-- Using source at /vcpkg/buildtrees/libb2/src/6f75badf85-b3a5d32c8d.clean
-- Getting CMake variables for x64-android-dbg
-- Getting CMake variables for x64-android-rel
-- Generating configure for x64-android
-- Finished generating configure for x64-android
-- Configuring x64-android-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: /usr/bin/bash -c "V=1 ./../src/6f75badf85-b3a5d32c8d.clean/configure --host=x86_64-linux-android \"ax_cv_check_cflags___O3=no\" \"--disable-silent-rules\" \"--verbose\" \"--disable-shared\" \"--enable-static\" \"--prefix=/project/host/vcpkg-android/vcpkg_installed/x64-android/debug\" \"--bindir=\\${prefix}/../tools/libb2/debug/bin\" \"--sbindir=\\${prefix}/../tools/libb2/debug/sbin\" \"--libdir=\\${prefix}/lib\" \"--includedir=\\${prefix}/../include\" \"--datarootdir=\\${prefix}/share/libb2\""
    Working Directory: /vcpkg/buildtrees/libb2/x64-android-dbg
    Error code: 77
    See logs for more information:
      /vcpkg/buildtrees/libb2/config-x64-android-dbg-config.log
      /vcpkg/buildtrees/libb2/config-x64-android-dbg-out.log
      /vcpkg/buildtrees/libb2/config-x64-android-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_make.cmake:832 (vcpkg_execute_required_process)
  buildtrees/versioning_/versions/libb2/7c113e12089453e4e2cf2bbf67ad1f0b80a133a8/portfile.cmake:14 (vcpkg_configure_make)
  scripts/ports.cmake:170 (include)


error: building libb2:x64-android failed with: BUILD_FAILED
Elapsed time to handle libb2:x64-android: 7 s
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+libb2
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?title=[libb2]+Build+error&body=Copy+issue+body+from+%2Fproject%2Fhost%2Fvcpkg-android%2Fvcpkg_installed%2Fvcpkg%2Fissue_body.md
```

```
$ cat /vcpkg/buildtrees/libb2/config-x64-android-dbg-config.log
This file contains any messages produced by compilers while
running configure, to aid debugging if configure makes a mistake.

It was created by libb2 configure 0.98.1, which was
generated by GNU Autoconf 2.71.  Invocation command line was

  $ ./../src/6f75badf85-b3a5d32c8d.clean/configure --host=x86_64-linux-android ax_cv_check_cflags___O3=no --disable-silent-rules --verbose --disable-shared --enable-static --prefix=/project/host/vcpkg-android/vcpkg_installed/x64-android/debug '--bindir=${prefix}/../tools/libb2/debug/bin' '--sbindir=${prefix}/../tools/libb2/debug/sbin' '--libdir=${prefix}/lib' '--includedir=${prefix}/../include' '--datarootdir=${prefix}/share/libb2'

## --------- ##
## Platform. ##
## --------- ##

hostname = 056b827c9e52
uname -m = x86_64
uname -r = 5.4.0-110-generic
uname -s = Linux
uname -v = #124-Ubuntu SMP Thu Apr 14 19:46:19 UTC 2022

/usr/bin/uname -p = x86_64
/bin/uname -X     = unknown

/bin/arch              = x86_64
/usr/bin/arch -k       = unknown
/usr/convex/getsysinfo = unknown
/usr/bin/hostinfo      = unknown
/bin/machine           = unknown
/usr/bin/oslevel       = unknown
/bin/universe          = unknown

PATH: /vcpkg/
PATH: /usr/local/sbin/
PATH: /usr/local/bin/
PATH: /usr/sbin/
PATH: /usr/bin/
PATH: /sbin/
PATH: /bin/
PATH: /vcpkg/downloads/tools/ninja/1.10.2-linux/
PATH: /bin/


## ----------- ##
## Core tests. ##
## ----------- ##

configure:2532: looking for aux files: config.guess config.sub ltmain.sh compile missing install-sh
configure:2545:  trying ./../src/6f75badf85-b3a5d32c8d.clean/
configure:2574:   ./../src/6f75badf85-b3a5d32c8d.clean/config.guess found
configure:2574:   ./../src/6f75badf85-b3a5d32c8d.clean/config.sub found
configure:2574:   ./../src/6f75badf85-b3a5d32c8d.clean/ltmain.sh found
configure:2574:   ./../src/6f75badf85-b3a5d32c8d.clean/compile found
configure:2574:   ./../src/6f75badf85-b3a5d32c8d.clean/missing found
configure:2556:   ./../src/6f75badf85-b3a5d32c8d.clean/install-sh found
configure:2706: checking for a BSD-compatible install
configure:2779: result: /usr/bin/install -c
configure:2790: checking whether build environment is sane
configure:2845: result: yes
configure:2901: checking for x86_64-linux-android-strip
configure:2933: result: /android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/bin/llvm-strip
configure:3004: checking for a race-free mkdir -p
configure:3048: result: /usr/bin/mkdir -p
configure:3055: checking for gawk
configure:3090: result: no
configure:3055: checking for mawk
configure:3076: found /usr/bin/mawk
configure:3087: result: mawk
configure:3098: checking whether make sets $(MAKE)
configure:3125: result: no
configure:3151: checking whether make supports nested variables
configure:3169: result: no
configure:3333: checking for x86_64-linux-android-gcc
configure:3365: result: /android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/bin/clang
configure:3763: checking for C compiler version
configure:3772: /android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/bin/clang --version >&5
Android (10552028, +pgo, +bolt, +lto, -mlgo, based on r487747d) clang version 17.0.2 (https://android.googlesource.com/toolchain/llvm-project d9f89f4d16663d5012e5c09495f3b30ece3d2362)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/bin
configure:3783: $? = 0
configure:3772: /android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/bin/clang -v >&5
Android (10552028, +pgo, +bolt, +lto, -mlgo, based on r487747d) clang version 17.0.2 (https://android.googlesource.com/toolchain/llvm-project d9f89f4d16663d5012e5c09495f3b30ece3d2362)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/bin
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/11
Selected GCC installation: /usr/lib/gcc/x86_64-linux-gnu/11
Candidate multilib: .;@m64
Selected multilib: .;@m64
configure:3783: $? = 0
configure:3772: /android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/bin/clang -V >&5
clang-17: error: argument to '-V' is missing (expected 1 value)
clang-17: error: no input files
configure:3783: $? = 1
configure:3772: /android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/bin/clang -qversion >&5
clang-17: error: unknown argument '-qversion'; did you mean '--version'?
clang-17: error: no input files
configure:3783: $? = 1
configure:3772: /android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/bin/clang -version >&5
clang-17: error: unknown argument '-version'; did you mean '--version'?
clang-17: error: no input files
configure:3783: $? = 1
configure:3803: checking whether the C compiler works
configure:3825: /android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/bin/clang -g -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -Wformat -Werror=format-security -fPIC -fno-limit-debug-info --target=x86_64-none-linux-android21 -DANDROID -D_FORTIFY_SOURCE=2 --sysroot=/android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/sysroot -avoid-version -L/project/host/vcpkg-android/vcpkg_installed/x64-android/debug/lib --target=x86_64-none-linux-android21 -static-libstdc++ -Wl,--build-id=sha1 -Wl,--no-rosegment -Wl,--no-undefined-version -Wl,--fatal-warnings -Wl,--no-undefined -Qunused-arguments --sysroot=/android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/sysroot conftest.c -latomic -lm >&5
clang: error: unknown argument: '-avoid-version'
configure:3829: $? = 1
configure:3869: result: no
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME "libb2"
| #define PACKAGE_TARNAME "libb2"
| #define PACKAGE_VERSION "0.98.1"
| #define PACKAGE_STRING "libb2 0.98.1"
| #define PACKAGE_BUGREPORT "contact@blake2.net"
| #define PACKAGE_URL "https://blake2.net"
| #define PACKAGE "libb2"
| #define VERSION "0.98.1"
| /* end confdefs.h.  */
| 
| int
| main (void)
| {
| 
|   ;
|   return 0;
| }
configure:3874: error: in `/vcpkg/buildtrees/libb2/x64-android-dbg':
configure:3876: error: C compiler cannot create executables
See `config.log' for more details

## ---------------- ##
## Cache variables. ##
## ---------------- ##

ac_cv_env_CC_set=set
ac_cv_env_CC_value=/android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/bin/clang
ac_cv_env_CFLAGS_set=set
ac_cv_env_CFLAGS_value='-g -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -Wformat -Werror=format-security -fPIC -fno-limit-debug-info'
ac_cv_env_CPPFLAGS_set=set
ac_cv_env_CPPFLAGS_value='--target=x86_64-none-linux-android21 -DANDROID -D_FORTIFY_SOURCE=2 --sysroot=/android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/sysroot'
ac_cv_env_LDFLAGS_set=set
ac_cv_env_LDFLAGS_value='-avoid-version -L/project/host/vcpkg-android/vcpkg_installed/x64-android/debug/lib --target=x86_64-none-linux-android21 -static-libstdc++ -Wl,--build-id=sha1 -Wl,--no-rosegment -Wl,--no-undefined-version -Wl,--fatal-warnings -Wl,--no-undefined -Qunused-arguments --sysroot=/android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/sysroot'
ac_cv_env_LIBS_set=set
ac_cv_env_LIBS_value='-latomic -lm'
ac_cv_env_LT_SYS_LIBRARY_PATH_set=
ac_cv_env_LT_SYS_LIBRARY_PATH_value=
ac_cv_env_build_alias_set=
ac_cv_env_build_alias_value=
ac_cv_env_host_alias_set=set
ac_cv_env_host_alias_value=x86_64-linux-android
ac_cv_env_target_alias_set=
ac_cv_env_target_alias_value=
ac_cv_path_install='/usr/bin/install -c'
ac_cv_path_mkdir=/usr/bin/mkdir
ac_cv_prog_AWK=mawk
ac_cv_prog_CC=/android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/bin/clang
ac_cv_prog_STRIP=/android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/bin/llvm-strip
ac_cv_prog_make_make_set=no
am_cv_make_support_nested_variables=no
ax_cv_check_cflags___O3=no

## ----------------- ##
## Output variables. ##
## ----------------- ##

ACLOCAL='${SHELL} '\''/vcpkg/buildtrees/libb2/src/6f75badf85-b3a5d32c8d.clean/missing'\'' aclocal-1.16'
AMDEPBACKSLASH=''
AMDEP_FALSE=''
AMDEP_TRUE=''
AMTAR='$${TAR-tar}'
AM_BACKSLASH='\'
AM_DEFAULT_V='1'
AM_DEFAULT_VERBOSITY='1'
AM_V='1'
AR='/android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/bin/llvm-ar'
AUTOCONF='${SHELL} '\''/vcpkg/buildtrees/libb2/src/6f75badf85-b3a5d32c8d.clean/missing'\'' autoconf'
AUTOHEADER='${SHELL} '\''/vcpkg/buildtrees/libb2/src/6f75badf85-b3a5d32c8d.clean/missing'\'' autoheader'
AUTOMAKE='${SHELL} '\''/vcpkg/buildtrees/libb2/src/6f75badf85-b3a5d32c8d.clean/missing'\'' automake-1.16'
AWK='mawk'
B2_LIBRARY_VERSION='1:4:0'
CC='/android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/bin/clang'
CCDEPMODE=''
CFLAGS='-g -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -Wformat -Werror=format-security -fPIC -fno-limit-debug-info'
CPPFLAGS='--target=x86_64-none-linux-android21 -DANDROID -D_FORTIFY_SOURCE=2 --sysroot=/android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/sysroot'
CSCOPE='cscope'
CTAGS='ctags'
CYGPATH_W='echo'
DEFS=''
DEPDIR=''
DLLTOOL=''
DSYMUTIL=''
DUMPBIN=''
ECHO_C=''
ECHO_N='-n'
ECHO_T=''
EGREP=''
ETAGS='etags'
EXEEXT=''
FGREP=''
GREP=''
INSTALL_DATA='${INSTALL} -m 644'
INSTALL_PROGRAM='${INSTALL}'
INSTALL_SCRIPT='${INSTALL}'
INSTALL_STRIP_PROGRAM='$(install_sh) -c -s'
LD='/android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/bin/ld.lld'
LDFLAGS='-avoid-version -L/project/host/vcpkg-android/vcpkg_installed/x64-android/debug/lib --target=x86_64-none-linux-android21 -static-libstdc++ -Wl,--build-id=sha1 -Wl,--no-rosegment -Wl,--no-undefined-version -Wl,--fatal-warnings -Wl,--no-undefined -Qunused-arguments --sysroot=/android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/sysroot'
LIBOBJS=''
LIBS='-latomic -lm'
LIBTOOL=''
LIPO=''
LN_S=''
LTLIBOBJS=''
LT_SYS_LIBRARY_PATH=''
MAKEINFO='${SHELL} '\''/vcpkg/buildtrees/libb2/src/6f75badf85-b3a5d32c8d.clean/missing'\'' makeinfo'
MANIFEST_TOOL=''
MKDIR_P='/usr/bin/mkdir -p'
NM='/android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/bin/llvm-nm'
NMEDIT=''
OBJDUMP='/android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/bin/llvm-objdump'
OBJEXT=''
OPENMP_CFLAGS=''
OTOOL64=''
OTOOL=''
PACKAGE='libb2'
PACKAGE_BUGREPORT='contact@blake2.net'
PACKAGE_NAME='libb2'
PACKAGE_STRING='libb2 0.98.1'
PACKAGE_TARNAME='libb2'
PACKAGE_URL='https://blake2.net'
PACKAGE_VERSION='0.98.1'
PATH_SEPARATOR=':'
RANLIB='/android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/bin/llvm-ranlib'
SED=''
SET_MAKE='MAKE=make'
SHELL='/bin/bash'
SIMD_FLAGS=''
STRIP='/android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/bin/llvm-strip'
USE_FAT_FALSE=''
USE_FAT_TRUE=''
USE_SSE_FALSE=''
USE_SSE_TRUE=''
VERSION='0.98.1'
ac_ct_AR=''
ac_ct_CC=''
ac_ct_DUMPBIN=''
am__EXEEXT_FALSE=''
am__EXEEXT_TRUE=''
am__fastdepCC_FALSE=''
am__fastdepCC_TRUE=''
am__include=''
am__isrc=' -I$(srcdir)'
am__leading_dot='.'
am__nodep=''
am__quote=''
am__tar='$${TAR-tar} chof - "$$tardir"'
am__untar='$${TAR-tar} xf -'
bindir='${prefix}/../tools/libb2/debug/bin'
build=''
build_alias=''
build_cpu=''
build_os=''
build_vendor=''
datadir='${datarootdir}'
datarootdir='${prefix}/share/libb2'
docdir='${datarootdir}/doc/${PACKAGE_TARNAME}'
dvidir='${docdir}'
exec_prefix='NONE'
host='x86_64-linux-android'
host_alias='x86_64-linux-android'
host_cpu=''
host_os=''
host_vendor=''
htmldir='${docdir}'
includedir='${prefix}/../include'
infodir='${datarootdir}/info'
install_sh='${SHELL} /vcpkg/buildtrees/libb2/src/6f75badf85-b3a5d32c8d.clean/install-sh'
libdir='${prefix}/lib'
libexecdir='${exec_prefix}/libexec'
localedir='${datarootdir}/locale'
localstatedir='${prefix}/var'
mandir='${datarootdir}/man'
mkdir_p='$(MKDIR_P)'
oldincludedir='/usr/include'
pdfdir='${docdir}'
pkgconfigdir=''
prefix='/project/host/vcpkg-android/vcpkg_installed/x64-android/debug'
program_transform_name='s,x,x,'
psdir='${docdir}'
runstatedir='${localstatedir}/run'
sbindir='${prefix}/../tools/libb2/debug/sbin'
sharedstatedir='${prefix}/com'
sysconfdir='${prefix}/etc'
target_alias=''

## ----------- ##
## confdefs.h. ##
## ----------- ##

/* confdefs.h */
#define PACKAGE_NAME "libb2"
#define PACKAGE_TARNAME "libb2"
#define PACKAGE_VERSION "0.98.1"
#define PACKAGE_STRING "libb2 0.98.1"
#define PACKAGE_BUGREPORT "contact@blake2.net"
#define PACKAGE_URL "https://blake2.net"
#define PACKAGE "libb2"
#define VERSION "0.98.1"

configure: exit 77
```
```
$ cat /vcpkg/buildtrees/libb2/config-x64-android-dbg-err.log 
configure: error: in `/vcpkg/buildtrees/libb2/x64-android-dbg':
configure: error: C compiler cannot create executables
See `config.log' for more details
```
```
$ cat /vcpkg/buildtrees/libb2/config-x64-android-dbg-out.log 
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for x86_64-linux-android-strip... /android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/bin/llvm-strip
checking for a race-free mkdir -p... /usr/bin/mkdir -p
checking for gawk... no
checking for mawk... mawk
checking whether make sets $(MAKE)... no
checking whether make supports nested variables... no
checking for x86_64-linux-android-gcc... /android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/bin/clang
checking whether the C compiler works... no
```

**Comment by AugP** [comment]
Converting this to draft for now since this seems to depend on a PR in the main vcpkg repo.

**Comment by janisozaur** [comment]
The PR in question is now merged

**Comment by dg0yt** [comment]
> The PR in question is now merged

Still, it takes a number of port updates to actual have vcpgk CI pass with r26 activated: https://github.com/microsoft/vcpkg/pull/35851.

File: vcpkg/users/platforms/android.md [hunk 1]
```diff
@@ -58,11 +58,11 @@ RUN \
 
 # Download Android NDK
 RUN \
-  wget https://dl.google.com/android/repository/android-ndk-r25c-linux.zip && \
-  unzip android-ndk-r25c-linux.zip && \
-  rm -rf android-ndk-r25c-linux.zip
+  wget https://dl.google.com/android/repository/android-ndk-r26b-linux.zip && \
+  unzip android-ndk-r26b-linux.zip && \
+  rm -rf android-ndk-r26b-linux.zip
```
Comment (janisozaur) [inline vcpkg/users/platforms/android.md:63]:
```suggestion
  wget https://dl.google.com/android/repository/android-ndk-r26c-linux.zip && \
  unzip android-ndk-r26c-linux.zip && \
  rm -rf android-ndk-r26c-linux.zip
```

File: vcpkg/users/platforms/android.md [hunk 2]
```diff
@@ -58,11 +58,11 @@ RUN \
...
 
-ENV ANDROID_NDK_HOME /android-ndk-r25c
+ENV ANDROID_NDK_HOME /android-ndk-r26b
```
Comment (janisozaur) [inline vcpkg/users/platforms/android.md:65]:
```suggestion
ENV ANDROID_NDK_HOME /android-ndk-r26c
```

**Comment by dg0yt** [comment]
You probably also want to change the PR title,

IMO vcpkg needs CI with r26 before this can be merged.

**Comment by vicroms** [comment]
Converted to draft until the relevant PRs are merged.

**Comment by dg0yt** [comment]
https://github.com/microsoft/vcpkg/pull/35851 is now using NDK r26c.

**Comment by AugP** [comment]
Made a new PR #356 updating to r26d and adding a few misc fixes. Added credit to @janisozaur for this initial PR, but closing this one as new one supersedes it.

**Pull request closed without merging**